pkg/populator/populator.go

```diff
@@ -16,11 +16,19 @@ import (
 // P is an interface that we can use to read from and to write to the kube config file.
 type P interface {
 	PopulateKubeConfig(project string) error
-	SetKubeConfigFile()
-	GetKubeConfigFile() string
 }
 
-
+//New instantiates a new P interface using the conf parameter. It can return a env, endpoint or oidc populator.
+func New(conf, kubeConfigFile string) (P, error) {
+	switch conf {
+	case "oidc":
+		return newOIDC(kubeConfigFile), nil
+	case "endpoint":
+		return newEndpoint(kubeConfigFile), nil
+	default:
```

Comment:
Having the env populator as the default should be determined in the command options, at this level we need to return an error that such a populator is not supported. Probably just a ErrNoSuchPopulator for now?